code/game/mecha/medical/odysseus.dm

```diff
@@ -10,37 +10,37 @@
 	internal_damage_threshold = 35
 	deflect_chance = 15
 	step_energy_drain = 6
-	var/obj/item/clothing/glasses/hud/health/mech/hud
-
-	New()
-		..()
-		hud = new /obj/item/clothing/glasses/hud/health/mech(src)
-		return
+	var/builtin_hud_user = 0
 
 	moved_inside(var/mob/living/carbon/human/H as mob)
 		if(..())
-			if(H.glasses)
-				occupant_message("<font color='red'>[H.glasses] prevent you from using [src] [hud]</font>")
+			if(H.glasses && istype(H.glasses, /obj/item/clothing/glasses/hud))
+				occupant_message("<span class='warning'>Your [H.glasses] prevent you from using the built-in medical hud.</span>")
 			else
-				H.glasses = hud
+				var/datum/atom_hud/data/human/medical/advanced/A = huds[DATA_HUD_MEDICAL_ADVANCED]
+				A.add_hud_to(H)
+				builtin_hud_user = 1
 			return 1
 		else
 			return 0
 
 	pilot_mmi_hud(var/mob/living/carbon/brain/pilot)
-		pilot.regular_hud_updates()
-		process_med_hud(pilot, 1)
+		//pilot.regular_hud_updates()
+		//process_med_hud(pilot, 1)
+		var/datum/atom_hud/data/human/medical/advanced/A = huds[DATA_HUD_MEDICAL_ADVANCED]
+		A.add_hud_to(pilot)
+		builtin_hud_user = 1
 		return ..()
 
 	go_out()
-		if(ishuman(occupant))
+		if(ishuman(occupant) && builtin_hud_user)

```

Comment:
Ah, caution! The pilot for odysseus mechs might be an MMI, in which case their mobtype will be /mob/living/carbon/brain - I'm not completely familiar with the new HUD code, but looking at this I suspect if you were to remove an MMI from an odysseus, it wouldn't lose its medical interface. That, and pilot_mmi_hud is called every tick of Life() of the brain, so you might want to reshape this so it's only called when the MMI is placed into the mech.